### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by chakapega**

### CODE REVIEW

File: src/constants/language.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const ENGLISH = "en";
```
Comment (ArtyomBobrovnik) [inline src/constants/language.ts]: I think you can use a `Enum` here

File: src/features/Dashboard/Dashboard.actions.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import * as constants from "./Dashboard.constants";
+import { AppThunkDispatch } from "store/types";
+import { INDEX_OF_FIRST_ELEMENT } from "constants/common";
+import { LanguageType } from "store/language/types";
+
+export const getNowPlayingMovies = (activeLanguage: LanguageType) => async (dispatch: AppThunkDispatch) => {
+  try {
+    dispatch({ type: constants.GET_NOW_PLAYING_MOVIES_START });
+
+    const nowPlayingMovies = await api.getNowPlayingMovies(activeLanguage);
```
Comment (ArtyomBobrovnik) [inline src/features/Dashboard/Dashboard.actions.ts]: Is it possible to provide limit to api, to fetch only `5`?

File: src/features/Dashboard/Dashboard.actions.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    const fiveNowPlayingMovies = nowPlayingMovies.slice(INDEX_OF_FIRST_ELEMENT, constants.NUMBER_OF_FILMS_IN_LIST);
```
Comment (ArtyomBobrovnik) [inline src/features/Dashboard/Dashboard.actions.ts]: better to not use word `five` in constant name, cause when you change `NUMBER_OF_FILMS_IN_LIST ` constant this name will be wrong

File: src/features/Dashboard/Dashboard.component.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  const activeLanguage = useAppSelector((state) => state.language.activeLanguage);
+  const {
+    isLoadingNowPlayingMovies,
+    nowPlayingMovies,
+    isLoadingUpcomingMovies,
+    upcomingMovies,
+    isLodaingPopularMovies,
+    popularMovies,
+  } = useAppSelector((state) => state.dashboard);
+  const getNowPlayingMoviesAvailable = !nowPlayingMovies.length && !isLoadingNowPlayingMovies;
```
Comment (ArtyomBobrovnik) [inline src/features/Dashboard/Dashboard.component.tsx]: To my mind it's strange name, I do not understand what it means

File: src/features/Dashboard/Dashboard.api.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+  const response = await fetch(
+    `${THE_MOVIE_DB_BASE_URL}/upcoming?api_key=${REACT_APP_THE_MOVIE_DB_KEY}&language=${countryCode}&page=1`
+  );
+  const { results } = await response.json();
+
+  return results;
+};
+
+export const getPopularMovies = async (activeLanguage: LanguageType) => {
```
Comment (ArtyomBobrovnik) [inline src/features/Dashboard/Dashboard.api.ts]: All this 3 methods different only in path, but we duplicate all strings

File: src/features/Dashboard/Dashboard.component.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  const getUpcomingMoviesAvailable = !upcomingMovies.length && !isLoadingUpcomingMovies;
+  const getPopularMoviesAvailable = !popularMovies.length && !isLodaingPopularMovies;
+
+  useEffect(() => {
+    if (getNowPlayingMoviesAvailable) {
+      dispatch(actions.getNowPlayingMovies(activeLanguage));
+    }
+  }, [getNowPlayingMoviesAvailable, activeLanguage, dispatch]);
```
Comment (ArtyomBobrovnik) [inline src/features/Dashboard/Dashboard.component.tsx]: What if list loaded, and after that use change language? I expect that system show him new result.

File: src/features/MovieDetails/MovieDetails.component.tsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  return (
+    <>
+      {movieDetails && (
+        <Container>
+          <h3>{movieDetails.title}</h3>
+          <p>{movieDetails.overview}</p>
+          <Container>
+            <Row>
+              <Col>
+                <img src={`${IMAGE_TMDB_BASE_URL}/${movieDetails.poster_path}`} alt='poster' />
```
Comment (ArtyomBobrovnik) [inline src/features/MovieDetails/MovieDetails.component.tsx]: Better to move this url generation to separate function

File: src/features/MovieDetails/MovieDetails.component.tsx [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+import { EMPTY_MOVIE_DETAILS_VALUE } from "./MovieDetails.constants";
+import { useAppSelector } from "hooks";
+import { getMovieDetails } from "./MovieDetails.api";
+import { IMAGE_TMDB_BASE_URL } from "constants/api";
+
+export const MovieDetails = () => {
+  const { id } = useParams<{ id: string }>();
+  const activeLanguage = useAppSelector((state) => state.language.activeLanguage);
+  const [movieDetails, setMovieDetails] = useState(EMPTY_MOVIE_DETAILS_VALUE);
```
Comment (ArtyomBobrovnik) [inline src/features/MovieDetails/MovieDetails.component.tsx]: Why we put dashboard lists in redux store, but not put this data?

File: src/features/Header/LanguageSwitcher/index.tsx [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+import { SET_LANGUAGE } from "store/language/actionTypes";
+import { RESET_MOVIES } from "features/Dashboard/Dashboard.constants";
+
+export const LanguageSwitcher = () => {
+  const activeLanguage = useAppSelector((state) => state.language.activeLanguage);
+  const dispatch = useAppDispatch();
+
+  const handleChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    dispatch({ type: SET_LANGUAGE, payload: event.target.value });
+    dispatch({ type: RESET_MOVIES });
```
Comment (ArtyomBobrovnik) [inline src/features/Header/LanguageSwitcher/index.tsx]: Why change of language reset movies?

File: src/features/Dashboard/Dashboard.actions.ts (hunk 2 above)
Comment (chakapega) [reply src/features/Dashboard/Dashboard.actions.ts]: No. 
Screenshot from 2021-08-08 01-31-11

File: src/features/Dashboard/Dashboard.component.tsx (hunk 4 above)
Comment (chakapega) [reply src/features/Dashboard/Dashboard.component.tsx]: Renamed to nowPlayingMoviesRequestAvailable, upcomingMoviesRequestAvailable, popularMoviesRequestAvailable

File: src/features/Dashboard/Dashboard.api.ts (hunk 5 above)
Comment (chakapega) [reply src/features/Dashboard/Dashboard.api.ts]: Replaced with one method getMovies

File: src/features/Dashboard/Dashboard.component.tsx (hunk 6 above)
Comment (chakapega) [reply src/features/Dashboard/Dashboard.component.tsx]: System show new result

File: src/features/MovieDetails/MovieDetails.component.tsx (hunk 7 above)
Comment (chakapega) [reply src/features/MovieDetails/MovieDetails.component.tsx]: Moved

**Pull request merged**